tests/unit/DependencyInjection/DependencyAware.class.test.js

```diff
@@ -5,10 +5,27 @@ const getEvent = require('../../mocks/aws/event.json');
 const getContext = require('../../mocks/aws/context.json');
 
 describe('DependencyInjection/DependencyAwareClass', () => {
-  const dependencyInjectionClass = new DependencyInjection({}, getEvent, getContext);
-  const dependencyAwareClass = new DependencyAware(dependencyInjectionClass);
+  describe('getContainer', () => {
+    const dependencyInjectionClass = new DependencyInjection({}, getEvent, getContext);
+    const dependencyAwareClass = new DependencyAware(dependencyInjectionClass);
 
-  it('should instantiate and be able to get the dependency injection container', () => {
-    expect(dependencyAwareClass.getContainer()).toEqual(dependencyInjectionClass);
+    it('should instantiate and be able to get the dependency injection container', () => {
+      expect(dependencyAwareClass.getContainer()).toEqual(dependencyInjectionClass);
+    });
+  });
+
+  describe('definitions', () => {
+    describe('Returns the provided definitions', () => {
+      [
+        [{}, undefined],
+        [{ DEFINITIONS: 1 }, 1],
+      ].forEach((testCase) => {
+        it(`With configuration: ${testCase[0]}`, () => {
+          const di = new DependencyInjection(testCase[0]);
+          const service = new DependencyAware(di);
+          expect(service.definitions).toEqual(testCase[1]);
+        });
```

Comment:
Rather than accessing the test case items by index, could you destructure the array? e.g.

```js
const [config, expectedDefinitions] = testCase;
it(`With configuration: ${config}`, () => {
  // ...
```

It's not immediately obvious what the structure of the list of test cases is, so this would also serve to document that.

Same applies for the list in `DependencyInjection.class.test.js`.

Reply:
1f5a20dd404db99c65ca0adba0212ab6ec33d94e

Reply:
Thanks, forgot this could be done in a function's argument list 🙂